packages/devtools/client/components/ComponentsGraph.vue

```diff
@@ -99,9 +102,12 @@ const data = computed<Data>(() => {
 
     const isGrayedOut = searchDebounced.value && !rel.id.toLowerCase().includes(searchDebounced.value.toLowerCase())
 
+    const label = path.split('/').splice(-1)[0].replace(/\.\w+$/, '')
```

Comment:
Shouldn't we use an utility function to infer a better name from the path, instead of relying on the selected state?

Reply:
That's interesting! But I'm not sure what the best approach is to extract the component name from the path.

Could you review my function if I implement it like this?

```ts
function getComponentName(path: string) {
  const splitPath = path.split('/');
  const lastChunkPath = splitPath.splice(-1)[0].replace(/\.\w+$/, '');

  if (lastChunkPath === 'index') {
    return splitPath.splice(-2)[0].replace(/\.\w+$/, '')
  }

  return lastChunkPath
}
```

If my code is incorrect or if you have a better approach, please let me know.